File: backend/app/Enums/PaymentMethod.php
<?php

namespace App\Enums;

enum PaymentMethod: string
{
    case Cash = 'cash';
    case Wallet = 'wallet';
    case Card = 'card';

    public function label(): string
    {
        return match ($this) {
            self::Cash => 'Cash',
            self::Wallet => 'Wallet',
            self::Card => 'Card',
        };
    }
}

File: packages/taxigo_core/lib/domain/enums/payment_method.dart
enum PaymentMethod {
  cash('cash'),
  wallet('wallet'),
  card('card');

  const PaymentMethod(this.value);

  final String value;

  static PaymentMethod fromString(String? raw) {
    return PaymentMethod.values.firstWhere(
      (method) => method.value == raw,
      orElse: () => PaymentMethod.cash,
    );
  }

  String get displayKey => switch (this) {
        PaymentMethod.cash => 'cash',
        PaymentMethod.wallet => 'wallet',
        PaymentMethod.card => 'card',
      };
}
